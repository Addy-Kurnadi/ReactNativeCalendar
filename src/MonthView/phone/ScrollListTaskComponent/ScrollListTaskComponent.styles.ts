import { StyleSheet, TextStyle, ViewStyle } from "react-native"

interface Style {
    touchableContainer: ViewStyle
    viewContainer: ViewStyle
    textStyleName: TextStyle
    textStyleTask: TextStyle
}

export default StyleSheet.create<Style>({
    touchableContainer: {
        flexDirection: "column",
        justifyContent: "center"
    },
    viewContainer: {
        paddingLeft: 4
    },
    textStyleName: {
        color: "#a9a9a9",
        fontWeight: "bold",
        fontSize: 12,
    },
    textStyleTask: {
        color: "#a9a9a9",
        fontSize: 12,
    }
})
