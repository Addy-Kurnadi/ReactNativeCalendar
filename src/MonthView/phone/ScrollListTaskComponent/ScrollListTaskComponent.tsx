import * as React from "react"
import { Text, TouchableOpacity, View } from "react-native"
import styles from "./ScrollListTaskComponent.styles"

interface ComponentProps {
    name: string
    task: string
    color: string
    height: number
    nameFontSize: number
    handleOnPress: () => void
}

export const DayTaskComponent = (props: ComponentProps) => {
    return (
        <TouchableOpacity
            onPress={props.handleOnPress}
            style={[
                styles.touchableContainer,
                {
                    height: props.height,
                    backgroundColor: props.color
                }
            ]}
        >
            <View style={styles.viewContainer}>
                <Text style={[styles.textStyleName, {fontSize: props.nameFontSize}]}>
                    {props.name}
                </Text>
                <Text style={[styles.textStyleTask, {fontSize: props.nameFontSize - 2}]}>
                    {props.task}
                </Text>
            </View>
        </TouchableOpacity>
    )
}
