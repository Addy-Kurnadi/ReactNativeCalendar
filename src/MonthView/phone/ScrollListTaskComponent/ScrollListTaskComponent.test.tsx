import * as React from "react"
import { create } from "react-test-renderer"
import { ScrollListTaskComponent } from "./ScrollListTaskComponent"

describe("<ScrollListTaskComponent />", () => {

    it("should render with given props", () => {
        const component = create(
            <ScrollListTaskComponent
                name={"Adam"}
                task={"Sleep"}
                color={"pink"}
                height={60}
                width={60}
                handleOnPress={jest.fn()}
            />).toJSON()
        expect(component).toMatchSnapshot()
    })
})
