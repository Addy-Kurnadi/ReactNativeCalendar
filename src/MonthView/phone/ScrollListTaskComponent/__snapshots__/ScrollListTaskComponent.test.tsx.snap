// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScrollListTaskComponent /> should render with given props 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "pink",
      "flex": 1,
      "flexDirection": "column",
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "paddingLeft": 4,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontWeight": "bold",
            "marginBottom": 1,
          },
          Object {
            "fontSize": 16,
          },
        ]
      }
    >
      Adam
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "marginTop": 1,
          },
          Object {
            "fontSize": 14,
          },
        ]
      }
    >
      Sleep
    </Text>
  </View>
</View>
`;
